### PR TITLE
Zep: Authentication; other bugfixes/updates; bump SDK version

### PR DESCRIPTION
**Pull request by danielchalef**
- zep auth; auth in example; fix for lastn
- if summary exists, add it to message history
- update class names
- bump zep-js version

### CODE REVIEW

**Comment by rsharath** [comment]
@nfcampos / @jacoblee93  - this change is ready for merge, appreciate you taking a look. Thanks!

File: langchain/src/memory/zep.ts [hunk 1]
```diff
@@ -51,15 +55,17 @@ export class ZepMemory extends BaseChatMemory implements ZepMemoryInput {
...
+    this.zepClient = new ZepClient(this.baseURL, fields.apiKey);
   }
 
   get memoryKeys() {
     return [this.memoryKey];
   }
 
   async loadMemoryVariables(values: InputValues): Promise<MemoryVariables> {
-    const lastN = values.lastN ?? 10;
+    // use either lastN provided by developer or undefined to use the
```
Comment (jacoblee93) [inline langchain/src/memory/zep.ts:66]: Will this default to 10 for current users?

File: langchain/package.json [hunk 2]
```diff
@@ -445,7 +445,7 @@
     "@clickhouse/client": "^0.0.14",
     "@faker-js/faker": "^7.6.0",
     "@getmetal/metal-sdk": "^4.0.0",
-    "@getzep/zep-js": "^0.3.1",
+    "@getzep/zep-js": "^0.4.1",
```
Comment (jacoblee93) [inline langchain/package.json:448]: Update the peer dep as well

**Review by jacoblee93 — CHANGES_REQUESTED** [review]
The integration test is now failing for me - can you fix?

**Comment by danielchalef** [comment]
This is a bug fix. The number of results returned should default to the message window size set on the server, not 10.

I’ll take a look at why the integration test is failing.

File: langchain/package.json (hunk 2 above)
Comment (danielchalef) [reply langchain/package.json:460]: Fixed

**Comment by danielchalef** [comment]
@jacoblee93 I've updated the peer dep, improved the robustness of the retriever, and added an additional test. Tests pass locally. Can you let me know if the CI tests pass? I can't see the results here.

As an aside, I have to manually modify the `test.skip` to `test` in to run them using `yarn test:single`. Is there a better way to force them to run from the CLI?

File: langchain/src/memory/zep.ts (hunk 1 above)
Comment (danielchalef) [reply langchain/src/memory/zep.ts:66]: This is a bug fix. The number of results returned should default to the message window size set on the server, not 10.

**Comment by jacoblee93** [comment]
> @jacoblee93 I've updated the peer dep, improved the robustness of the retriever, and added an additional test. Tests pass locally. Can you let me know if the CI tests pass? I can't see the results here.
> 
> As an aside, I have to manually modify the `test.skip` to `test` in to run them using `yarn test:single`. Is there a better way to force them to run from the CLI?

The alternative is not skipping them at all by adding a container setup here:
https://github.com/hwchase17/langchainjs/blob/main/test-int-deps-docker-compose.yml

**Review by jacoblee93 — APPROVED** [review]

**Comment by jacoblee93** [comment]
Fixed out of date yarn.lock and merged in with #1811

**Comment by danielchalef** [comment]
Thanks, @jacoblee93 !

**Pull request closed without merging**